packages/tests/src/e2e/privacyPlugin/privacyPlugin.spec.ts

```diff
@@ -0,0 +1,104 @@
+    // Navigate to our page.
+    await page.goto(`${url}/index.html?context_bundler=${bundler}`);
+    await page.waitForSelector('body');
+};
+
+describe('Privacy Plugin', () => {
+    // Build our fixture project.
+    beforeAll(async ({ publicDir, bundlers, suiteName }) => {
+        const source = path.resolve(__dirname, 'project');
+        const destination = path.resolve(publicDir, suiteName);
+        await verifyProjectBuild(source, destination, bundlers);
+    });
+
+    test('Should have set global variables in the helper', async ({
+        page,
+        bundler,
+        browserName,
+        suiteName,
+        devServerUrl,
+    }) => {
+        const errors: string[] = [];
+        const testBaseUrl = `${devServerUrl}/${suiteName}`;
+
+        // Listen for errors on the page.
+        page.on('pageerror', (error) => errors.push(error.message));
+        page.on('response', async (response) => {
+            if (!response.ok()) {
+                const url = response.request().url();
+                const prefix = `[${bundler} ${browserName} ${response.status()}]`;
+                errors.push(`${prefix} ${url}`);
+            }
+        });
+
+        await userFlow(testBaseUrl, page, bundler);
+
+        const ddAllow = await page.evaluate(() => {
+            return (globalThis as any).$DD_ALLOW;
+        });
+
+        expect(ddAllow).toBeDefined();
+        expect(errors).toEqual([]);
+    });
+
+    test('Should trigger the callback when new scripts are loaded', async ({
+        page,
+        bundler,
+        browserName,
+        suiteName,
+        devServerUrl,
+    }) => {
+        // skip for webpack4 because of dynamic import
+        if (bundler === 'webpack4') {
+            test.skip();
+        }
+
+        const errors: string[] = [];
+        const testBaseUrl = `${devServerUrl}/${suiteName}`;
+
+        // Listen for errors on the page.
+        page.on('pageerror', (error) => errors.push(error.message));
+        page.on('response', async (response) => {
+            if (!response.ok()) {
+                const url = response.request().url();
+                const prefix = `[${bundler} ${browserName} ${response.status()}]`;
+                errors.push(`${prefix} ${url}`);
+            }
+        });
+
+        await userFlow(testBaseUrl, page, bundler);
+
+        await page.evaluate(() => {
+            const globalAny: any = globalThis;
+            if (!globalAny.$DD_ALLOW_OBSERVERS) {
+                globalAny.$DD_ALLOW_OBSERVERS = new Set<() => void>();
+            }
+            globalAny.$DD_ALLOW_OBSERVERS.add(() => {
+                console.log('DD_ALLOW observer triggered');
+            });
+        });
+
+        const button = page.getByTestId('load-script');
+        const waitForLog = page.waitForEvent('console', (msg) =>
+            msg.text().includes('DD_ALLOW observer triggered'),
+        );
+        await button.click();
+        expect(await waitForLog).toBeTruthy();
```

Comment:
Since you're gathering errors, you can do an assertion on them:
```suggestion
        expect(await waitForLog).toBeTruthy();
        expect(errors).toEqual([]);
```

Otherwise, remove the gathering mechanism.